### ISSUE TITLE
--extra-index flag throws error

### ISSUE THREAD

**Issue by matteosantama** [cjolowicz/nox-poetry#314]
My CI starts to fail if I bump `nox-poetry` from 0.8.1 to 0.8.2 with the following error

```
...
 File "/usr/local/lib/python3.8/site-packages/nox_poetry/sessions.py", line 209, in export_requirements
    constraints = to_constraints(self.poetry.export())
  File "/usr/local/lib/python3.8/site-packages/nox_poetry/sessions.py", line 88, in to_constraints
    return "\n".join(_to_constraints())
  File "/usr/local/lib/python3.8/site-packages/nox_poetry/sessions.py", line 84, in _to_constraints
    constraint = to_constraint(requirement, line)
  File "/usr/local/lib/python3.8/site-packages/nox_poetry/sessions.py", line 69, in to_constraint
    raise RuntimeError(f"line {line}: {requirement_string!r}: {error}")
RuntimeError: line 1: '--extra-index-url https://gitlab.com/api/v4/projects/20895990/packages/pypi/simple': 
Parse error at "'--extra-'": Expected W:(abcd...)

````

**Comment by cjolowicz** [comment]
Thanks for reporting!

We should ignore requirement lines starting with ˋ-ˋ as it is not a valid way to start a package name, and must be a pip option. The function is ˋ_to_constraintˋ in the sessions module of you're interested in submitting a PR.

**Comment by cjolowicz** [comment]
@matteosantama A fix was released in [0.8.3], can you try?

[0.8.3]: https://github.com/cjolowicz/nox-poetry/releases/tag/v0.8.3

**Comment by RomainBrault** [comment]
Thanks your patch sloved part of my error. Now I have a new one which seem related to this one:

Here is a log

```bash
nox > Running session pre-commit                                                                                                                                                                                   
nox > Creating virtual environment (virtualenv) using python3.9 in .cache/pre-commit                                                                                                                               
nox > poetry build --format=wheel                                                                                                                                                                                  
nox > poetry export --format=requirements.txt --dev --without-hashes                                                                                                                                               
nox > Session pre-commit raised exception RuntimeError('line 2: \'\': Parse error at "\'\'": Expected W:(abcd...)')                                                                                                
Traceback (most recent call last):                                                                                                                                                                                 
  File "/home/romain/Git/ppt/.cache/tests-3-9/lib/python3.9/site-packages/packaging/requirements.py", line 113, in __init__                                                                                        
    req = REQUIREMENT.parseString(requirement_string)                                                                                                                                                              
  File "/home/romain/Git/ppt/.cache/tests-3-9/lib/python3.9/site-packages/pyparsing.py", line 1955, in parseString                                                                                                 
    raise exc                                                                                                                                                                                                      
  File "/home/romain/Git/ppt/.cache/tests-3-9/lib/python3.9/site-packages/pyparsing.py", line 3250, in parseImpl                                                                                                   
    raise ParseException(instring, loc, self.errmsg, self)                                                                                                                                                         
pyparsing.ParseException: Expected W:(abcd...)  (at char 0), (line:1, col:1)                                                                                                                                       
                                                                                                                                                                                                                   
During handling of the above exception, another exception occurred:                                                                                                                                                
                                                                                                                                                                                                                   
Traceback (most recent call last):                                                                                                                                                                                 
  File "/home/romain/Git/ppt/.cache/tests-3-9/lib/python3.9/site-packages/nox_poetry/sessions.py", line 67, in to_constraint                                                                                       
    requirement = Requirement(requirement_string)                                                                                                                                                                  
  File "/home/romain/Git/ppt/.cache/tests-3-9/lib/python3.9/site-packages/packaging/requirements.py", line 115, in __init__                                                                                        
    raise InvalidRequirement(                                                                                                                                                                                      
packaging.requirements.InvalidRequirement: Parse error at "''": Expected W:(abcd...)                                                                                                                               
                                                                                                                                                                                                                   
During handling of the above exception, another exception occurred:                                                                                                                                                
                                                                                                                                                                                                                   
Traceback (most recent call last):                                                                                                                                                                                 
  File "/home/romain/Git/ppt/.cache/tests-3-9/lib/python3.9/site-packages/nox/sessions.py", line 549, in execute                                                                                                   
    self.func(session)                                                                                                                                                                                             
  File "/home/romain/Git/ppt/.cache/tests-3-9/lib/python3.9/site-packages/nox/_decorators.py", line 53, in __call__                                                                                                
    return self.func(*args, **kwargs)                                                                                                                                                                              
  File "/home/romain/Git/ppt/.cache/tests-3-9/lib/python3.9/site-packages/nox_poetry/sessions.py", line 42, in wrapper                                                                                             
    function(proxy, *_args, **_kwargs)                                                                                                                                                                             
  File "/home/romain/Git/ppt/.cache/tests-3-9/lib/python3.9/site-packages/typeguard/__init__.py", line 912, in wrapper                                                                                             
    retval = func(*args, **kwargs)                                                                                                                                                                                 
  File "/tmp/pytest-of-romain/pytest-35/test_bake_project_tiss_inner_o0/cookies/bake00/Awesome Thales Project/noxfile.py", line 225, in precommit                                                                  
    session.poetry.installroot(distribution_format=nox_poetry.WHEEL)                                                                                                                                               
  File "/home/romain/Git/ppt/.cache/tests-3-9/lib/python3.9/site-packages/nox_poetry/sessions.py", line 170, in installroot                                                                                        
    requirements = self.export_requirements()                                                                                                                                                                      
  File "/home/romain/Git/ppt/.cache/tests-3-9/lib/python3.9/site-packages/nox_poetry/sessions.py", line 209, in export_requirements                                                                                
    constraints = to_constraints(self.poetry.export())                                                                                                                                                             
  File "/home/romain/Git/ppt/.cache/tests-3-9/lib/python3.9/site-packages/nox_poetry/sessions.py", line 88, in to_constraints                                                                                      
    return "\n".join(_to_constraints())                                                                                                                                                                            
  File "/home/romain/Git/ppt/.cache/tests-3-9/lib/python3.9/site-packages/nox_poetry/sessions.py", line 84, in _to_constraints                                                                                     
    constraint = to_constraint(requirement, line)                                                                                                                                                                  
  File "/home/romain/Git/ppt/.cache/tests-3-9/lib/python3.9/site-packages/nox_poetry/sessions.py", line 69, in to_constraint                                                                                       
    raise RuntimeError(f"line {line}: {requirement_string!r}: {error}")                                                                                                                                            
RuntimeError: line 2: '': Parse error at "''": Expected W:(abcd...)                                                                                                                                                
nox > Session pre-commit failed. 
```

**Comment by matteosantama** [comment]
@cjolowicz Unfortunately the patch doesn't totally resolve the issue. I am getting the same error as @RomainBrault 

My `pyproject.toml` looks like

```
[[tool.poetry.source]]
name = "gitlab_pypi"
url = "https://gitlab.com/api/v4/projects/XXX/packages/pypi/simple"
secondary = false
```

**Comment by cjolowicz** [comment]
@matteosantama @RomainBrault thanks for the input and the PR!

Another fix was released in [0.8.4], please let me know if you still encounter any problems.

[0.8.4]: https://github.com/cjolowicz/nox-poetry/releases/tag/v0.8.4
